bin/update_mirror.py

```diff
@@ -92,6 +92,10 @@ def test(hosts,tag,arch):
         except urllib2.HTTPError,e:
             #no such repo on this host..
             log("\tURL caught while processing url:"+url+" "+str(e))
+        except urllib2.URLError,e:
+            # Error contacting the host. Remove it from the mirrorhosts for this run.
+            log("\tExcluding host due to connection error for url:"+url+" "+str(e))
+            mirrorhosts.remove(host)
```

Comment:
Ah, nice.  Off hand i don't know if `Name or service not known` is the only `urllib2.URLError` that might be valid here.  But yeah if a host cant be contacted, no use re-attempting for each url.